infra/terraform/base_stacks_2/dns_records.tf

```diff
@@ -0,0 +1,11 @@
+resource "aws_cloudformation_stack" "dns-records" {
+    name = "platform-dns-records"
+
+    template_body = file("../../templates/dns-records/template.yaml")
+
+    parameters = {
+        Environment = var.environment
+    }
+
+    depends_on = [aws_cloudformation_stack.dns-records]
```

Comment:
This "depends_on" is not required as it is only referencing itself. It might not even pass `terraform validation` 

Terraform's job is to find the dependencies between different resources. Typically this is done implicitly if a resource depends on the output of another (for example passing a security group name to a lambda vpc config, terraform will make the security group first) 

Sometime you require resources to be made in a particular order without them having an explicit reference between the two. In cases like this you would add the depends on to enforce that relationship. 

An example might be two cloud fromation stacks, with one using an `!ImportValue`  of an output from another. There is no relationship between the two as defined in terraform but there is a relationship between the two in AWS. Here you would use the depends_on clause to ensure they are deployed correctly. 

Since this terraform module has a single resource to deploy, the depends_on has no functionality. The dependency tree here is managed by the fact we won't deploy `base_stacks_2` until we have completed our dns record deployment into the dev platform account managing the route 53 record.